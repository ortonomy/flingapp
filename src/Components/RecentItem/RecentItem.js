import React from 'react'
import './RecentItem.css'

const RecentItem = (props) => {
  const icon = `fa fa-${props.icon}`
  return (
    <div className='RecentItem'>
      <i className={icon} aria-hidden="true"></i>
      <span><a href={props.linkTarget}>{props.displayText}</a></span>
    </div>
  )
}

export default RecentItem;
