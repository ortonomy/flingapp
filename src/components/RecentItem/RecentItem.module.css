.RecentItem {
  padding: 1em 0.5em;
  border-bottom: solid 1px rgba(0,0,0,0.3);
}

.RecentItem i {
  height: 24px;
  width: 24px;
  color: #4990e2;
  font-size: 16px;
  border-radius: 50%;
  border: solid 0.5px #4990e2;
  text-align: center;
  line-height: 20px;
}

.RecentItem a {
  text-transform: capitalize;
}
