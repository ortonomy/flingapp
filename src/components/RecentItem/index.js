import React from 'react'
import recent from './RecentItem.module.css'

const RecentItem = (props) => {
  const icon = `fa fa-${props.icon}`
  return (
    <div className={recent.RecentItem}>
      <i className={icon} aria-hidden="true"></i>
      <span><a href={props.linkTarget}>{props.displayText}</a></span>
    </div>
  )
}

export default RecentItem;
