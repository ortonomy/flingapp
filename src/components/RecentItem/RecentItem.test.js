import React from 'react';
import RecentItem from '../RecentItem';

it('renders without crashing', () => {
  const wrapper = shallow(<RecentItem />);
  expect(wrapper).toMatchSnapshot();
});
