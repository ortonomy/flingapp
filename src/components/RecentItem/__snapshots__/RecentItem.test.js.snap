// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecentItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          aria-hidden="true"
          className="fa fa-undefined"
/>,
        <span>
          <a
                    href={undefined}
          />
</span>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "className": "fa fa-undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "href": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "a",
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            aria-hidden="true"
            className="fa fa-undefined"
/>,
          <span>
            <a
                        href={undefined}
            />
</span>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "fa fa-undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "href": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
